tests/unittests/test_datasource/test_scaleway.py

```diff
@@ -403,3 +404,25 @@ def test_network_config_existing(self, m_get_cmdline, fallback_nic):
 
         netcfg = self.datasource.network_config
         self.assertEqual(netcfg, '0xdeadbeef')
+
+    @mock.patch('cloudinit.sources.DataSourceScaleway.net.find_fallback_nic')
+    @mock.patch('cloudinit.util.get_cmdline')
+    def test_network_config_unset(self, m_get_cmdline, fallback_nic):
+        """
+        _network_config will be set to sources.UNSET after the first run.
```

Comment:
that comment doesn't seem right.
what do you mean 'first run'?  after `cloud-init local` or second boot for the instance?

Either way, I think we should have cached the response somewhere.

maybe collect-logs on a failed instance  and attach?

Reply:
Well, looks like GH doesn't want my logs tarball. I'll try to find another way to make it available.